app/src/main/java/org/hackillinois/android/view/navigationdrawer/ScannerFragment.kt

```diff
@@ -0,0 +1,195 @@
+            var userId: String = getUserIdFromQrString(lastSuccessfullyScannedUser)
+
+
+            Log.d("ScanEvent", "User ID: ${userId}")
+            Log.d("ScanEvent", "Event: ${eventName}")
+
+            // Check-in is a special event in the HackIllinois API
+            if (eventName == "Check In") {
+                var override = staffOverrideSwitch.isChecked
+                var hasCheckedIn = true
+                var hasPickedUpSwag = true
+                var checkIn = CheckIn(userId, override, hasCheckedIn, hasPickedUpSwag)
+
+                viewModel.checkInUser(checkIn)
+
+            } else {
+                var userEventPair = UserEventPair(eventName, userId)
+
+                viewModel.markUserAsAttendingEvent(userEventPair)
+            }
+        }
+    }
+
+    /**
+     * A callback that is called when the status of the last scan request changes.
+     */
+    fun processLastScan(lastScanWasSuccessful: Boolean?) {
+        when (lastScanWasSuccessful) {
+            false -> {
+                Toast.makeText(context, "Try again!", Toast.LENGTH_LONG).show()
+            }
+            true -> {
+                lastSuccessfullyScannedUser = viewModel.lastUserIdScannedIn.value.toString()
+                Toast.makeText(context, "Success: ${lastSuccessfullyScannedUser}",
+                        Toast.LENGTH_LONG).show()
+                staffOverrideSwitch.isChecked = false
+            }
+            null -> {
+                return
+            }
+        }
+
+    }
+
+    /**
+     * Extracts the userId from the HackIllinois QR String URI.
+     * Given an example URI "hackillinois://user?userid=github0000001"
+     * it is enough to grab everything after the last (and only) equals sign.
+     *
+     * If the URI becomes more complex, we can use actual URI decoders.
+     * @return the userId from a QR String
+     */
+    private fun getUserIdFromQrString(qrString: String): String {
+        var splitOnEquals = qrString.split("=")
+        return splitOnEquals.last()
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+
+        if (requestCode == PERMISSIONS_REQUEST_ACCESS_CAMERA) {
+            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                qrScanner.decodeContinuous(onQrCodeScan)
+            }
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        // Inflate the layout for this fragment
+        var view = inflater.inflate(org.hackillinois.android.R.layout.fragment_scanner, container, false)
+
+        view.eventListView.onItemSelectedListener = viewModel.onEventSelected
+
+        // Ensure the Camera permission is granted
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
```

Comment:
I'm not a huge fan of writing long boolean expressions on multiple lines. Instead you could make boolean variables for both parts of this expression.

Reply:
Sure, will do!

Reply:
Android Studio makes it very complex to do this without Annotations and other stuff, since this stuff doesn't apply for Android versions that didn't require dynamic permissions requests.